Examples/BookingSystem.AspNetCore/Controllers/OpenBookingController.cs

```diff
@@ -198,5 +198,34 @@ public IActionResult TestInterfaceAction([FromServices] IBookingEngine bookingEn
             }
         }
 
+        /// <summary>
```

Comment:
initially I just had one method:

```cs
[Route("{*url}")]
public IActionResult UnknownEndpointWithBody([FromServices] IBookingEngine bookingEngine)
{
    var error = new OpenBookingException(new UnknownOrIncorrectEndpointError());
    return error.ErrorResponseContent.GetContentResult();
}
```

When the test suite test sent a PUT request (as detailed here https://github.com/openactive/openactive-test-suite/pull/162/files#diff-50a90b10bcd259c04240b7a1d63bf106R17-R25), OpenActive.Server.NET returned a 415 Unsupported Media Type. After a quick bit of looking around, I thought this might be because there was a `[FromBody]` arg missing and therefore the route was being declared as not expecting JSON requests.

So I added another route with a `[FromBody]` arg, in the hopes that this would be called for JSON requests and the other for non-JSON requests (e.g. GET requests).

However, I'm still getting the 415 for the PUT request and now a 500 for GET. Just wondering if you knew about this off the top of your head. I can continue to dig but I don't really know ASP.NET

Reply:
This is likely the one we want: "Status Code Pages With Re Execute":
https://www.devtrends.co.uk/blog/handling-404-not-found-in-asp.net-core

This allows us to customise 404 (UnknownOrIncorrectEndpointError), 405 (MethodNotAllowed), and a catch-all 500 (InternalApplicationError), to ensure they all return JSON.

We should probably add 415 to the booking spec so that one can be passed through too?

Reply:
(have added a proposal: https://github.com/openactive/open-booking-api/issues/145)